spec/recurly/resource_spec.rb

```diff
@@ -84,6 +84,22 @@ def to_param
       end
     end
 
+    describe ".find_each" do
+      it "must accept a block" do
+        stub_api_request(:get, 'resources?per_page=50') { XML[200][:index] }
+        results = []
+        resource.find_each { |r| r.must_be_instance_of resource ; results << r }
+        results.wont_be_empty
+      end
+
+      it "must allow chaining of iterator methods without passing a block" do
+        stub_api_request(:get, 'resources?per_page=50') { XML[200][:index] }
+        results = []
+        resource.find_each.to_a.map.each { |r| r.must_be_instance_of resource ; results << r }
```

Comment:
A single one of them would do.
On Fri, Aug 28, 2015 at 8:09 PM Arron Mabrey <arron@mabreys.com> wrote:

> Literally nothing but show/prove that it can chain. :-)
> On Fri, Aug 28, 2015 at 7:43 PM Benjamin Eckel <notifications@github.com>
> wrote:
>
>> In spec/recurly/resource_spec.rb
>> <https://github.com/recurly/recurly-client-ruby/pull/209#discussion_r38254647>
>> :
>>
>> > @@ -84,6 +84,22 @@ def to_param
>> >        end
>> >      end
>> >
>> > +    describe ".find_each" do
>> > +      it "must accept a block" do
>> > +        stub_api_request(:get, 'resources?per_page=50') { XML[200][:index] }
>> > +        results = []
>> > +        resource.find_each { |r| r.must_be_instance_of resource ; results << r }
>> > +        results.wont_be_empty
>> > +      end
>> > +
>> > +      it "must allow chaining of iterator methods without passing a block" do
>> > +        stub_api_request(:get, 'resources?per_page=50') { XML[200][:index] }
>> > +        results = []
>> > +        resource.find_each.to_a.map.each { |r| r.must_be_instance_of resource ; results << r }
>>
>> Why the to_a the map and the each ?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/recurly/recurly-client-ruby/pull/209/files#r38254647>
>> .
>>
>